core/src/Instance/InstanceDiscoveryResponse.cs

```diff
@@ -35,5 +35,8 @@ internal sealed class InstanceDiscoveryResponse : OAuth2ResponseBase
     {
         [DataMember(Name = "tenant_discovery_endpoint", IsRequired = false)]
         public string TenantDiscoveryEndpoint { get; set; }
+
+        [DataMember(Name = "metadata")]
```

Comment:
seems like this should be required. however the same above. Do you know why the tenent_discovery_endpoint is not required?